squish/src/test_data.rs

```diff
@@ -0,0 +1,202 @@
+    0x00, 0x24, 0x48, 0x6D, // row 0
+    0x91, 0xB6, 0xDB, 0xFF, // row 1
+    0x00, 0x24, 0x48, 0x6D, // row 2
+    0x91, 0xB6, 0xDB, 0xFF, // row 3
+];
+
+/// The same test pattern as BC2_GRAY, but with with 255/7 steps in the alpha values.
+/// Note that the alpha values are rounded down.
+pub const BC3_GRAY: TestDataSet = TestDataSet {
+    encoded: &[
+        0x24, 0xDB, 0x86, 0xC6, 0xE6, 0x86, 0xC6, 0xE6, // Alpha
+        0xFF, 0xFF, 0x00, 0x00, 0x44, 0x3D, 0x7C, 0x11, // Colour
+    ],
+    decoded: &add_custom_alpha_to_rgb(&expand_single_to_rgb(&GRAY_BLOCK_LUMA), &BC3_ALPHA_DECODED),
+};
+
+/// The same test pattern as BC2_COLOUR, but with 255/7 steps in the alpha values.
+/// Note that the alpha values are rounded down.
+pub const BC3_COLOUR: TestDataSet = TestDataSet {
+    encoded: &[
+        0x24, 0xDB, 0x86, 0xC6, 0xE6, 0x86, 0xC6, 0xE6, // Alpha
+        0xA9, 0xFC, 0x45, 0xFB, 0x00, 0xFF, 0x55, 0x55, // Colour
+    ],
+    decoded: &add_custom_alpha_to_rgb(&COLOUR_BLOCK_RGB, &BC3_ALPHA_DECODED),
+};
+
+/// The BC1_GRAY test pattern, but as BC4 with BC3 alpha encoding.
+pub const BC4_GRAY: TestDataSet = TestDataSet {
+    encoded: &[0x7F, 0x84, 0xF7, 0x6D, 0xE0, 0x07, 0xEC, 0xFB],
+    decoded: &add_alpha_to_rgb(
+        &expand_single_to_rgb(&[
+            0xFF, 0x00, 0xFF, 0x00, // row 0
+            0x00, 0x7F, 0x7F, 0xFF, // row 1
+            0xFF, 0x7F, 0x7F, 0x00, // row 2
+            0x00, 0xFF, 0x00, 0xFF, // row 3
+        ]),
+        0xFF,
+    ),
+};
+
+/// The BC1_GRAY test pattern for the red channel and the inverse pattern for the green channel.
+pub const BC5_GRAY: TestDataSet = TestDataSet {
+    encoded: &[
+        0x7F, 0x84, 0xF7, 0x6D, 0xE0, 0x07, 0xEC, 0xFB, // Alpha 0
+        0x7F, 0x84, 0xBE, 0x7F, 0xC0, 0x06, 0x7E, 0xDF, // Alpha 1
+    ],
+    decoded: &add_alpha_to_rgb(
+        &[
+            0xFF, 0x00, 0x00, 0x00, 0xFF, 0x00, 0xFF, 0x00, 0x00, 0x00, 0xFF, 0x00, // row 0
+            0x00, 0xFF, 0x00, 0x7F, 0x7F, 0x00, 0x7F, 0x7F, 0x00, 0xFF, 0x00, 0x00, // row 1
+            0xFF, 0x00, 0x00, 0x7F, 0x7F, 0x00, 0x7F, 0x7F, 0x00, 0x00, 0xFF, 0x00, // row 2
+            0x00, 0xFF, 0x00, 0xFF, 0x00, 0x00, 0x00, 0xFF, 0x00, 0xFF, 0x00, 0x00, // row 3
+        ],
+        0xFF,
+    ),
+};
+
+/// Expands an array with a single value per pixel to an array with this value expanded
+/// into the RGB channels.
+const fn expand_single_to_rgb(input: &[u8; 4 * 4]) -> [u8; 4 * 4 * 3] {
+    let mut output = [0u8; 4 * 4 * 3];
+    let mut i = 0;
+    // for loops are not available in const functions at the time of writing
+    while i < input.len() {
+        output[i * 3 + 0] = input[i]; // R
+        output[i * 3 + 1] = input[i]; // G
+        output[i * 3 + 2] = input[i]; // B
+        i += 1;
+    }
+    output
+}
+
+/// Adds to the RGB data the alpha value and returns the resulting RGBA data.
+const fn add_alpha_to_rgb(input: &[u8; 4 * 4 * 3], alpha: u8) -> [u8; 4 * 4 * 4] {
```

Comment:
This is redundant since you can just pass `&[alpha; 4*4]` to the generic fn. The dedicated fn for this case doesn't make it much shorter to write anyway.